pkg/services/ngalert/notifier/multiorg_alertmanager.go

```diff
@@ -72,14 +72,40 @@ func NewMultiOrgAlertmanager(cfg *setting.Cfg, configStore AlertingStore, orgSto
 		decryptFn:     decryptFn,
 		metrics:       m,
 		ns:            ns,
+		peer:          &NilPeer{},
 	}
+	if err := moa.setupClustering(cfg); err != nil {
+		return nil, err
+	}
+	return moa, nil
+}
 
-	clusterLogger := l.New("component", "cluster")
-	moa.peer = &NilPeer{}
+func (moa *MultiOrgAlertmanager) setupClustering(cfg *setting.Cfg) error {
+	clusterLogger := moa.logger.New("component", "clustering")
+	// Redis setup.
+	if cfg.UnifiedAlerting.HARedisAddr != "" {
+		redisPeer, err := newRedisPeer(redisConfig{
+			addr:     cfg.UnifiedAlerting.HARedisAddr,
+			name:     cfg.UnifiedAlerting.HARedisPeerName,
+			prefix:   cfg.UnifiedAlerting.HARedisPrefix,
+			password: cfg.UnifiedAlerting.HARedisPassword,
+			username: cfg.UnifiedAlerting.HARedisUsername,
+			db:       cfg.UnifiedAlerting.HARedisDB,
+		}, clusterLogger, moa.metrics.Registerer, cfg.UnifiedAlerting.HAPushPullInterval)
+		if err != nil {
+			return fmt.Errorf("unable to initialize redis: %w", err)
+		}
+		var ctx context.Context
+		ctx, moa.settleCancel = context.WithTimeout(context.Background(), 30*time.Second)
+		go redisPeer.Settle(ctx, cluster.DefaultGossipInterval*10)
+		moa.peer = redisPeer
+		return nil
+	}
+	// Memberlist setup.
 	if len(cfg.UnifiedAlerting.HAPeers) > 0 {
```

Comment:
What happens if the user configures _both_ memberlist peers and redis?

Reply:
Only Redis will start, as you can only have one cluster peer per Grafana instance.